fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/PredicateMultiMap.java

```diff
@@ -163,16 +148,101 @@ public Optional<CorrelationIdentifier> getParameterAliasOptional() {
                 return Optional.empty();
             }
-            final var predicateConjunctionPredicate = (PredicateWithValueAndRanges)this.queryPredicate;
+            final var predicateConjunctionPredicate = (PredicateWithValueAndRanges)queryPredicate;
             return Optional.of(Iterables.getOnlyElement(predicateConjunctionPredicate.getRanges()).asComparisonRange());
         }
 
         @Nonnull
         public Optional<QueryPlanConstraint> getConstraint() {
-            return constraint;
+            return constraintOptional;
+        }
+
+        @Nonnull
+        public static PredicateMapping regularMapping(@Nonnull final QueryPredicate queryPredicate,
+                                                      @Nonnull final QueryPredicate candidatePredicate,
+                                                      @Nonnull final CompensatePredicateFunction compensatePredicateFunction) {
+            return regularMapping(queryPredicate, candidatePredicate, compensatePredicateFunction, Optional.empty(), Optional.empty());
+        }
+
+        @Nonnull
+        public static PredicateMapping regularMapping(@Nonnull final QueryPredicate queryPredicate,
+                                                      @Nonnull final QueryPredicate candidatePredicate,
+                                                      @Nonnull final CompensatePredicateFunction compensatePredicateFunction,
+                                                      @Nonnull final Optional<CorrelationIdentifier> parameterAliasOptional,
+                                                      @Nonnull final Optional<QueryPlanConstraint> constraintOptional) {
+            return new PredicateMapping(queryPredicate, candidatePredicate, MappingKind.REGULAR_IMPLIES_CANDIDATE, compensatePredicateFunction, parameterAliasOptional, constraintOptional);
+        }
+
+        @Nonnull
+        public static PredicateMapping orTermMapping(@Nonnull final QueryPredicate queryPredicate,
+                                                     @Nonnull final QueryPredicate candidatePredicate,
+                                                     @Nonnull final CompensatePredicateFunction compensatePredicateFunction) {
+            return orTermMapping(queryPredicate, candidatePredicate, compensatePredicateFunction, Optional.empty(), Optional.empty());
+        }
+
+        @Nonnull
+        public static PredicateMapping orTermMapping(@Nonnull final QueryPredicate queryPredicate,
+                                                     @Nonnull final QueryPredicate candidatePredicate,
+                                                     @Nonnull final CompensatePredicateFunction compensatePredicateFunction,
+                                                     @Nonnull final Optional<CorrelationIdentifier> parameterAliasOptional,
+                                                     @Nonnull final Optional<QueryPlanConstraint> constraintOptional) {
```

Comment:
Is the constraint relevant for this type of mapping?

Reply:
Not really -- the worst that can happen is that we think an or term matches but then later on it doesn't. So it's not the end of the world.